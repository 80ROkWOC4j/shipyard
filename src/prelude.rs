pub use crate::get::GetComponent;
pub use crate::iter::{iterators, IntoIter, Shiperator};
pub use crate::not::Not;
pub use crate::remove::Remove;
pub use crate::run::System;
#[doc(hidden)]
pub use crate::run::SystemData;
pub use crate::sparse_set::{sort, sort::Sortable, View, ViewMut};
pub use crate::storage::{AllStorages, Entities, EntitiesMut, Key};
pub use crate::world::World;
pub use crate::Unique;
#[doc(hidden)]
#[cfg(feature = "proc")]
pub use shipyard_proc::system;

#[cfg(feature = "parallel")]
pub use crate::ThreadPool;
