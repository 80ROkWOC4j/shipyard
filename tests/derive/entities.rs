use shipyard::prelude::*;

#[system(Test1)]
fn run(_: Entities, _: EntitiesMut) {}

#[system(Test2)]
fn run(_: Entities, _: &mut Entities) {}

#[system(Test3)]
fn run(_: &Entities, _: EntitiesMut) {}

#[system(Test4)]
fn run(_: &Entities, _: &mut Entities) {}

fn main() {}
