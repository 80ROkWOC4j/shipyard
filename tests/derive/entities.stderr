error: Cannot borrow again a storage already borrowed exclusively, you may want to remove the shared borrow.
 --> $DIR/entities.rs:4:8
  |
4 | fn run(_: Entities, _: EntitiesMut) {}
  |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Cannot borrow again a storage already borrowed exclusively, you may want to remove the shared borrow.
 --> $DIR/entities.rs:7:8
  |
7 | fn run(_: Entities, _: &mut Entities) {}
  |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Cannot borrow again a storage already borrowed exclusively, you may want to remove the shared borrow.
  --> $DIR/entities.rs:10:8
   |
10 | fn run(_: &Entities, _: EntitiesMut) {}
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Cannot borrow again a storage already borrowed exclusively, you may want to remove the shared borrow.
  --> $DIR/entities.rs:13:8
   |
13 | fn run(_: &Entities, _: &mut Entities) {}
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
